File: cooldownbutton-sample/src/main/java/io/github/chncs23/cooldownbutton/sample/MainActivity.java
package io.github.chncs23.cooldownbutton.sample;

import android.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import io.github.chncs23.cooldownbutton.CooldownButton;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    final CooldownButton cooldownButton = (CooldownButton) findViewById(R.id.cooldown);
    cooldownButton.setDelayInSeconds(3);
    cooldownButton.setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View view) {
        Toast.makeText(MainActivity.this, "Clicked", Toast.LENGTH_SHORT).show();
      }
    });

    cooldownButton.setOnPostDelayListener(new CooldownButton.OnPostDelayListener() {
      @Override public void onPostDelay() {
        AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
        builder.setTitle("Title");
        builder.setMessage("hello");
        builder.show();
      }
    });
  }
}

File: cooldownbutton-sample/src/main/java/io/github/chncs23/cooldownbutton/sample/MainActivity.kt
package io.github.chncs23.cooldownbutton.sample

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import io.github.chncs23.cooldownbutton.CooldownButton

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button: CooldownButton = findViewById(R.id.cooldown) as CooldownButton
        button.setDelay(1)
        button.setOnClickListener {
            Toast.makeText(this, "Button Pressed", Toast.LENGTH_LONG).show()
        }

        button.setOnPostDelayListener(
                CooldownButton.OnPostDelayListener { Toast.makeText(this, "after delay", Toast.LENGTH_SHORT).show() })
    }
}
